test/overtone/sc/core_test.clj

```diff
@@ -1,48 +1,29 @@
 (ns overtone.sc.core-test
-  (:use
-   clojure.test
-   overtone.live
-   ;; TODO: this is not importable until the server is booted?
-   overtone.inst.synth)
   (:require
-   [overtone.sc.machinery.server.connection :refer [connection-info*]]
-   [overtone.config.log :as log]))
+    [clojure.test :refer [deftest is use-fixtures]]
+    [overtone.sc.server :refer [boot-server status kill-server]]
+    [overtone.sc.node :refer [group group-free]]
+    [overtone.sc.machinery.server.connection :refer [connection-info*]]
+    [overtone.music.time :refer [now]]
+    [overtone.config.log :as log]
+    [overtone.test-helper :as th]))
 
-(def ditty-notes [50 50 57 50 48 62 62 50])
-(def ditty-durs  [250 250 500 125 125 250 250 500])
-
-(defn play-seqs [inst notes durs]
-  (loop [notes notes
-         durs durs
-         t (now)]
-    (when (and notes durs)
-      ;; TODO: hit is not defined
-      ;; (hit t inst :note (first notes) :dur (first durs))
-      (recur (next notes) (next durs) (+ t (first durs))))))
+(use-fixtures :once th/ensure-server)
 
 (deftest boot-test
-  (try
-    (boot-server)
-    (Thread/sleep 500)
-    (is (not (nil? @connection-info*)))
-    (is (= 1 (:n-groups (status))))
-
-    ;; (play-seqs "sin" ditty-notes ditty-durs)
-    (Thread/sleep 3000)
-    (finally
-      (kill-server))))
+  (is (some? @connection-info*))
+  (is (some? (:n-groups (status)))))
 
 (def DEFAULT-GROUP 1)
 
 (deftest groups-test
-  (let [a (group :head DEFAULT-GROUP)
+  (let [init-groups (:n-groups (status))
+        a (group :head DEFAULT-GROUP)
         b (group :head DEFAULT-GROUP)
         c (group :head DEFAULT-GROUP)]
-    (is (= 4 (:n-groups (status))))
+    (is (= 3 (- (:n-groups (status)) init-groups)))
     (run! group-free [a b c])
-    (is (= 1 (:n-groups (status))))
-    ;; We should get the old ID again with the bitset allocator
-    (is (= a (group :head DEFAULT-GROUP)))))
```

Comment:
This test will never pass because we call `next-id` which always returns a fresh id.